applications/crossbar/src/modules_v2/cb_lists_v2.erl

```diff
@@ -0,0 +1,259 @@
+                                  ,{<<"last_name">>, wh_json:get_value(<<"lastname">>, JObj1)}
+                                  ,{<<"middle_name">>, wh_json:get_value(<<"middlename">>, JObj1)}
+                                  ,{<<"nicknames">>, [wh_json:get_value(<<"displayname">>, JObj1)]}])
+              ,JObj1),
+    JObj3 = set_org(JObj2, Context1),
+    JObj4 = set_photo(JObj3, Context1),
+    RespData = kzd_user:to_vcard(JObj4),
+    cb_context:set_resp_data(Context1, [RespData, "\n"]).
+
+-spec validate_doc(api_binary(), ne_binary(), cb_context:context()) -> cb_context:context().
+validate_doc(Id, Type, Context) ->
+    OnSuccess = fun(C) -> on_successfull_validation(Id, Type, C) end,
+    cb_context:validate_request_data(type_schema_name(Type), Context, OnSuccess).
+
+-spec delete(cb_context:context(), path_token()) -> cb_context:context().
+-spec delete(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+-spec delete(cb_context:context(), path_token(), path_token(), path_token()) -> cb_context:context().
+delete(Context, ListId) ->
+    delete(Context, ListId, ?ENTRIES),
+    Context1 = crossbar_doc:load(ListId, Context),
+    crossbar_doc:delete(Context1).
+delete(Context, _ListId, ?ENTRIES) ->
+    Docs = [wh_json:get_value(<<"id">>, Entry) || Entry <- cb_context:doc(Context)],
+    AccountDb = wh_util:format_account_id(cb_context:account_db(Context), 'encoded'),
+    %% do we need 'soft' delete as in crossbar_doc?
+    couch_mgr:del_docs(AccountDb, Docs),
+    Context.
+delete(Context, _ListId, ?ENTRIES, _EntryId) ->
+    crossbar_doc:delete(Context).
+
+-spec save(cb_context:context()) -> cb_context:context().
+-spec save(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+save(Context) ->
+    crossbar_doc:save(Context).
+save(Context, _) ->
+    save(Context).
+save(Context, _, ?ENTRIES) ->
+    save(Context).
+save(Context, _, ?ENTRIES, _) ->
+    save(Context).
+save(Context, _, ?ENTRIES, EntryId, ?PHOTO) ->
+    %% May be move code from cb_users_v2 to crossbar_doc?
+    cb_users_v2:post(Context, EntryId, ?PHOTO).
+
+-spec type_schema_name(ne_binary()) -> ne_binary().
+type_schema_name(<<"list">>) -> <<"lists">>;
+type_schema_name(<<"list_entry">>) -> <<"list_entries">>.
+
+-spec on_successfull_validation(api_binary(), ne_binary(), cb_context:context()) -> cb_context:context().
+on_successfull_validation('undefined', Type, Context) ->
+    Doc = wh_json:set_values([{<<"pvt_type">>, Type}], cb_context:doc(Context)),
+    cb_context:set_doc(Context, Doc);
+on_successfull_validation(Id, _Type, Context) ->
+    crossbar_doc:load_merge(Id, Context).
+
+-spec set_org(wh_json:object(), cb_context:context()) -> wh_json:object().
+set_org(JObj, Context) ->
+    case wh_json:get_value(<<"org">>
+                           ,cb_context:doc(crossbar_doc:load(wh_json:get_value(<<"list_id">>, JObj)
```

Comment:
This doesn't account for if crossbar_doc:load/2 fails. Break it into steps:

```erlang
set_org(JObj, Context) ->
    Context1 = crossbar_doc:load(...),
    set_org(JObj, Context1, cb_context:resp_status(Context1)).

set_org(JObj, Context, 'success') ->
    case wh_json:get_value(<<"org">>, cb_context:doc(Context)) of
        ...
    end;
set_org(JObj, _Context, _Status) ->
    lager:debug("failed to load list: ~s", [_Status]),
    JObj.
```

What you do in the error case is up to you (maybe throw?).